app/controllers/account/names_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Account
+  class NamesController < ApplicationController
+    before_action :redirect_if_name_exists
+
+    def edit
+      @name_form = NameForm.new(user: current_user).assign_form_values
+    end
+
+    def update
+      @name_form = NameForm.new(account_name_form_params(current_user))
+
+      if @name_form.submit
+        redirect_to root_path
+      else
+        render :edit
```

Comment:
I often forget to do this, but I think we usually try to return 422 for validation errors:
```suggestion
        render :edit, status: :unprocessable_entity
```